src/clj_s3_client/core.clj

```diff
@@ -62,7 +44,103 @@
       object-metadata
       (conj specific-setters user-metadata-setter))))
 
-(defn put-object [^AmazonS3Client client bucket-name object-key ^InputStream is options]
+(defn- suppress-not-found-exception [exception]
+  (when-not (= 404 (.getStatusCode exception))
+    (throw exception)))
+
+(defn create-client
+  "## (create-client)
+
+  Returns instance of AmazonS3Client which can be used with all other functions to access S3 resources.
+
+  An optional map of options can include any of the following keys:
+
+    :max-connections    - the max number of active connections for client.
+                          Defaults to 50.
+
+    :max-error-retry    - the number of retries to try before failing of http action.
+                          Defaults to 1.
+
+    :endpoint           - the endpoint which client will use.
+
+    :region             - the region which client will use.
+
+    :tcp-keep-alive     - set the keep connection alive flag for http connections.
+                          Defaults to false.
+
+  "
+  [& {:keys [max-connections max-error-retry endpoint region tcp-keep-alive]
+                        :or {max-connections 50 max-error-retry 1 tcp-keep-alive false}}]
+  (let [configuration (-> (ClientConfiguration.)
+                          (.withMaxErrorRetry max-error-retry)
+                          (.withMaxConnections max-connections)
+                          (.withTcpKeepAlive tcp-keep-alive))
+        client (AmazonS3Client. configuration)]
+    (when endpoint
+      (.setEndpoint client endpoint))
+    (if region
+      (.withRegion client (Regions/fromName region))
+      client)))
+
+(defn create-bucket
+  "## (create-bucket client \"my-awesome-bucket\")
+
+  Creates bucket by given name.
+  "
+  [^AmazonS3Client client bucket-name]
+  (.createBucket client bucket-name))
+
+(defn delete-bucket
+  "## (delete-bucket client \"my-not-so-awesome-bucket\")
+
+  Delete bucket by given name.
+  "
+  [^AmazonS3Client client bucket-name]
+  (.deleteBucket client bucket-name))
+
+(defn exists?-bucket [^AmazonS3Client client bucket-name]
```

Comment:
The function naming is a bit weird. `bucket-exists?` makes more sense. Feels somewhat out of place to have the question mark in the middle of function name.